src/exporters/warpten.rs

```diff
@@ -0,0 +1,366 @@
+                help: String::from("TCP port to join Warp10 on the host"),
+                long: String::from("port"),
+                short: String::from("p"),
+                required: false,
+                takes_value: true,
+            },
+        );
+        options.insert(
+            String::from("write-token"),
+            ExporterOption {
+                default_value: None,
+                help: String::from("Auth. token to write on Warp10"),
+                long: String::from("write-token"),
+                short: String::from("t"),
+                required: true,
+                takes_value: true,
+            },
+        );
+        options.insert(
+            String::from("step"),
+            ExporterOption {
+                default_value: Some(String::from("60")),
+                help: String::from("Time step between measurements, in seconds."),
+                long: String::from("step"),
+                short: String::from("S"),
+                required: true,
+                takes_value: true,
+            },
+        );
+        options.insert(
+            String::from("qemu"),
+            ExporterOption {
+                default_value: None,
+                help: String::from("Time step between measurements, in seconds."),
+                long: String::from("qemu"),
+                short: String::from("q"),
+                required: false,
+                takes_value: false,
+            },
+        );
+
+        options
+    }
+}
+
+impl Warp10Exporter {
+    pub fn new(mut sensor: Box<dyn Sensor>) -> Warp10Exporter {
+        if let Some(topo) = *sensor.get_topology() {
+            Warp10Exporter { topology: topo }
+        } else {
+            error!("Could'nt generate the Topology.");
+            panic!("Could'nt generate the Topology.");
+        }
+    }
+
+    pub fn iteration(
+        &mut self,
+        host: &str,
+        scheme: &str,
+        port: u16,
+        write_token: &str,
+        qemu: bool,
+    ) -> Result<warp10::Warp10Response, warp10::Error> {
+        let client = warp10::Client::new(&format!("{}://{}:{}", scheme, host, port))?;
+        let writer = client.get_writer(write_token.to_string());
+        self.topology
+            .proc_tracker
+            .clean_terminated_process_records_vectors();
+
+        debug!("Refreshing topology.");
+        self.topology.refresh();
+
+        let records = self.topology.get_records_passive();
+        let scaphandre_version = get_scaphandre_version();
+
+        let labels = vec![
+            warp10::Label::new("scaphandre_self_version", &scaphandre_version),
+            warp10::Label::new("agent", "scaphandre"),
+        ];
+
+        let mut data = vec![warp10::Data::new(
+            time::OffsetDateTime::now_utc(),
+            None,
+            String::from("scaph_self_version"),
+            labels.clone(),
+            warp10::Value::Double(scaphandre_version.parse::<f64>().unwrap()),
```

Comment:
Another thing we could factorize maybe. Maybe, we could leverage the Metric trait to send something whatever the type it will select the appropriate method (Double, Long, ...).

Reply:
I think you have a point here. This would be much cleaner. I don't yet imagine how it would integrate with the warp10 lib however.